examples/greeter-extension/index.js

```diff
@@ -5,7 +5,18 @@
 
 module.exports = require('./dist');
 
-if (require.main === module) {
-  const app = new module.exports.GreetingApplication();
-  app.main();
+async function main() {
+  if (require.main === module) {
+    const app = new module.exports.GreetingApplication({
+      rest: {host: '127.0.0.1', port: 3000},
+    });
+    const cli = await app.main();
+    if (!cli) {
+      console.log(
+        `The service is running at ${app.restServer.url}/greet/world`,
+      );
+    }
+  }
 }
+
+main();
```

Comment:
I find it suspicious that `main` is always executed. 

I believe we are using the following approach in other places:

```ts
if (require.main === module) {
  main().catch(err => {
    console.error(err);
    process.exit(1);
  });
}

async function main() {
  const app = new module.exports.GreetingApplication({
    // ...
  }
}
```